### DISCUSSION TITLE
Remote development, connection failed (RSA key)

### DISCUSSION THREAD

**Discussion by Awakened-Redstone** [zed-industries/zed#19992] [Feature Requests]
Hello, I have been looking forward for improving my remote development setup, moving from using Intellij trough xforwarding to something that has the UI running on my client. I decided to try zed for it as it has a remote development feature that seems to meet my needs. But during setup of the remote dev I encountered an error, when adding the host, the connection fails due to the required RSA key on my remote, giving me the message "Permission denied (publickey).. Please try again"

I tried providing the rsa file with `-i` but the result is the same. I am able to connect to my server with ssh on the terminal just fine. I am not asked to insert the passphrase at any point too.

I have checked the logs and all it says is `failed to connect: <user>@<host>: Permission denied (publickey).`

**Comment by alessio-locatelli** [comment]
I have a similar problem, but not sure if it is the same.

A working `ssh` command:

```bash
sshpass -p root_password ssh root@localhost -p 32768
```

But running `zed ssh://root@localhost:32768/~` or  `zed ssh://root:root_password@localhost:32768/~` gives an error:

![2024-10-31_10-00](https://github.com/user-attachments/assets/fb1f7541-ec62-44f8-9c30-a2c98cbd84bb)

**Comment by Awakened-Redstone** [comment]
I have opened an issue about this and I was able to resolve it by adding the `openbsd-netcat` package as referenced [here](https://github.com/zed-industries/zed/issues/20634#issuecomment-2484266400)

The package name mentioned is for Arch Linux, it may be named differently on other distros, for example `netcat-openbsd`
